### DISCUSSION TITLE
Request for pressure calibration option

### DISCUSSION THREAD

**Discussion by rolandbreedveld** [arendst/Tasmota#11339] [Ideas]
I use, a bme-280 with the tasmota-sensors version 9.3.1.1
It's nice that it's possible to calibrate power, voltage, temperature, humidity, etc.
But there is no command to calibrate air-pressure. like HumOffset and TempOffset.
is it possible to add it, or is there another way to do this?

**Comment by barbudor** [comment]
There is an option for stating your Altitude so the computation of SeaPressure is correct but I don't think this is what you are looking for.

What is you experience with Pressure that would justify the needed adjustment ?
So far BME280 pressure has been considered as an absolute correct value (within the accuracy of the datasheet of course).
Do you have access to a higher accurate Pressure sensor that would allow you to calibrate the BME280 measurement ?

**Reply by rolandbreedveld** [reply]
Hi Temperature on my BME280 is 5.5 degrees to high and pressure 15 hPa to low.
Yes, I have equipment which is high accurate.
for temperature it's easy to calibrate, would nice if it's also possible with pressure.

**Reply by sfromis** [reply]
Unless there are "hyperlocal" differences with ambient temperature (like being heated by a nearby ESP8266 radiator), a temperature 5.5 degrees too high indicates a faulty sensor. My working BME280 sensors have a less than 2 hPa deviation, a bit of this may be due to altitude differences only correctable to +/- 1 meter. I'd scrap sensors way wrong, as I'd not trust a stable linear deviation.

**Reply by rolandbreedveld** [reply]
> Unless there are "hyperlocal" differences with ambient temperature (like being heated by a nearby ESP8266 radiator), a temperature 5.5 degrees to high indicates a faulty sensor.

sensor is mounted at some distance from the other electronics, also outside the case.
I'll try another one, perhaps the aliexpress one's are not so good....

**Reply by sfromis** [reply]
Mine are all from Aliexpress sellers. I've however had a pretty bad batch of BMP280 (the version without humidity) ....

**Reply by rolandbreedveld** [reply]
> Mine are all from Aliexpress sellers. I've however had a pretty bad batch of BMP280 (the version without humidity) ....

Think I have the same bad batch, set of 10, all bad, also no humidity, ...

**Reply by sfromis** [reply]
Is the small "silver" sensor chip with a hole nearly square, or clearly longer on one side? The latter is BMP280. My bad batch had 40% not at all working, not even bad data.

**Reply by rolandbreedveld** [reply]
it's not square, but about 4x3 and the hole is on the right/lower side

**Reply by sfromis** [reply]
Clearly BMP280. BME280 has the hole center on one edge. If you thought you were buying the better BME280 sensors, you've been fleeced. Not saying that BMP280 is "bad" per se, but I suspect that some sellers make a bit of money selling rejects.

**Reply by rolandbreedveld** [reply]
waiting 4 the "real" bme's, hope this works better

**Reply by sfromis** [reply]
Good luck! You may need it ;-)

**Reply by rolandbreedveld** [reply]
> Good luck! You may need it ;-)

thanxzz, will inform if this is better !

**Reply by barbudor** [reply]
15 hPa is quite above the accuracy advertised in BME280 datasheet.
It's quite possible as not many ones have a high precision barometer at home anyone has noticed the possible error so far.
More over 15 hPA is generally not much a problem when you just want to estimate if tomorrow will have a better weather and looking only to variations 🤣

If it comes that such a PressureOffset is required, this should be easy to add.
Looking forward your results

**Reply by sfromis** [reply]
When I have something like 10 BME280 sensors (not all from the same batch) giving pressure values well within the specs sensitivity error of ±0.25%, I tend to believe in them not all erring significantly to one side, and trust that I have the "real" pressure within the deviation of less than 2 hPa I'm seeing. I think that I had one outlier deviating a bit more, till I replaced it. The specs "allow" for a deviation up to approx 5 hPa, if two sensors have max margin to opposite sides.

**Reply by Jason2866** [reply]
Dont buy the variant which supports 5V too. It is a design fail since there is voltage regulator onboard which heats the sensor.
